### PR TITLE
IAddressBook: add isShared and isSystemAddressBook

### PR DESCRIPTION
**Pull request by georgehrke**

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Looks great!

**Review by call-me-matt — APPROVED** [review]
perfect!

File: lib/public/IAddressBook.php [hunk 1]
```diff
@@ -107,5 +107,17 @@ public function getPermissions();
 		 * @since 5.0.0
 		 */
 		public function delete($id);
+
+		/**
+		 * @return bool
+		 * @since 20.0.0
+		 */
+		public function isShared(): bool;
```
Comment (call-me-matt) [inline lib/public/IAddressBook.php:115]: maybe add a comment to make clear this checks if the address book has been shared with the current user - and does not return if the address book has been shared with other users.

Comment (georgehrke) [reply lib/public/IAddressBook.php:118]: Done

**Pull request merged**